src/librados/AioCompletionImpl.h

```diff
@@ -85,10 +85,7 @@ struct librados::AioCompletionImpl {
     return wait_for_complete();
   }
   int is_complete() {
-    lock.Lock();
-    int r = complete;
-    lock.Unlock();
-    return r;
+    return complete;
```

Comment:
`complete` is updated in another thread by `finish()`, we need to add a memory barrier to make sure complete `complete` is synchronized before reading it. and POSIX mutex underlying `Mutex` ensure this. i don't see why we can remove the lock.